src/jupyterlabcontroller/models/domain/prepuller.py

```diff
@@ -5,6 +5,8 @@
 from dataclasses import dataclass, field
 from typing import Dict, List, Optional, TypeAlias
 
+from pydantic import BaseModel
```

Comment:
Isn't BaseModel a strong hint that it's not a domain model?  I had thought in general Pydantic was to be used for things that will be part of the external API, and Dataclass for internal models.

Reply:
This is going to be temporary anyway since I think we can test the same things through the API, but I went ahead and put this back to a dataclass so that we can keep that invariant.